tests/test_apiproxy.py

```diff
@@ -21,145 +19,126 @@ class TestAPIProxy(TestShared):
     API and API Proxy tests.
     """
 
-    @dastollervey_datasaver
-    def setUp(self):
+    @qrexec_datasaver
+    def setup_method(self):
         self.totp = pyotp.TOTP("JHCOGO7VCER3EJ4L")
         self.username = "journalist"
         self.password = "correct horse battery staple profanity oil chewy"
         self.server = "http://localhost:8081/"
         self.api = API(self.server, self.username, self.password, str(self.totp.now()), proxy=True)
+        try:
+            self.api.authenticate()
+        except BaseError as err:
+            raise AuthError(
+                "Could not obtain API token during test setup. "
+                "TOTP code may have expired or proxy may not be reachable. "
+                "Error was: {}".format(err.msg)
+            )
 
-        for i in range(3):
-            if os.path.exists("testtoken.json"):
-                token = load_auth()
-                self.api.token = token
-                self.api.update_auth_header()
-                break
-
-            # The following is True when we did a logout
-            if os.path.exists("logout.txt"):
-                os.unlink("logout.txt")
-                time.sleep(31)
-            # Now let us try to login
-            try:
-                self.api = API(
-                    self.server, self.username, self.password, str(self.totp.now()), proxy=True
-                )
-                self.api.authenticate()
-                with open("login.txt", "w") as fobj:
-                    fobj.write("in")
-            except Exception as err:
-                print(err)
-                time.sleep(31)
-                continue
-
-            save_auth(self.api.token)
-            break
-
+    @qrexec_datasaver
```

Comment:
adding the datasaver decorator here makes it so that the readme instructions are no longer accurate. on `main` you'll see that you can follow these steps from the readme:

> 2. [Skip if adding a new test] Delete the cassettes you wish to regenerate or just delete all json files by running:
> 
>     ```bash
>     rm data/*.json
>     ```
> 
> 3. Comment out the `@qrexec_datasaver` decorator above the test you want to generate a new cassette for or just generate all new cassettes by commenting out the decorator above all methods in the `test_apiproxy.py::TestAPIProxy` class.
> 
> 4. Make qrexec calls to the server and collect real response data:
> 
>     ```bash
>     make test TESTS=tests/test_apiproxy.py
>     ```

Step 3 is where things seem to go wrong, perhaps because of adding this decorator here?

Reply:
Good catch! I think it's because we're no longer conditionally loading a recorded token based on the presence of a `testtoken.json` file on disk, which means that re-running an individual test without commenting out `qrexec_datasaver` for `setup_method` will not work (because it'll replay the last token during that test's setup). Will verify and clarify.